pkg/resources/session_parameter.go

```diff
@@ -64,70 +68,59 @@ func CreateSessionParameter(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
-	// add quotes to value if it is a string
-	typeString := reflect.TypeOf("")
-	if reflect.TypeOf(parameterDefault.DefaultValue) == typeString {
-		value = fmt.Sprintf("'%s'", snowflake.EscapeString(value))
-	}
-
-	builder := snowflake.NewSessionParameter(key, value, db)
-
 	onAccount := d.Get("on_account").(bool)
 	user := d.Get("user").(string)
+	parameter := sdk.SessionParameter(key)
 
+	var err error
 	if onAccount {
-		builder.SetOnAccount(onAccount)
+		err := client.Parameters.SetSessionParameterForAccount(ctx, parameter, value)
+		if err != nil {
+			return err
+		}
 	} else {
 		if user == "" {
 			return fmt.Errorf("user is required if on_account is false")
```

Comment:
Can we move this logic to a validation function in the resource schema?

Reply:
I'm not sure if we can do that since we only check if it's empty if on_account is false. validation functions in resource schema cannot access another attribute value afaik